pkg/kapp/cmd/app/delete.go

```diff
@@ -70,8 +72,28 @@ func (o *DeleteOptions) Run() error {
 	}
 
 	if !exists {
-		o.ui.PrintLinef("%s", notExistsMsg)
-		return nil
+		if o.MigrationFlags.prevAppName != "" {
+			o.AppFlags.Name = o.MigrationFlags.prevAppName
+
+			app, supportObjs, err = Factory(o.depsFactory, o.AppFlags, o.ResourceTypesFlags, o.logger)
+			if err != nil {
+				return err
+			}
+
+			prevExists, prevNotExistsMsg, err := app.Exists()
```

Comment:
(No strong opinions)
```suggestion
			prevAppExists, prevAppNotExistsMsg, err := app.Exists()
```